src/query/server/query.go

```diff
@@ -471,7 +471,7 @@ func Run(runOpts RunOptions) RunResult {
 			logger.Fatal("error constructing etcd client", zap.Error(err))
 		}
 		logger.Info("setup noop storage backend with etcd")
-
+	// NB: all cases above are broken!! "downsampler" won't be initialized if the cases below are not executed.
```

Comment:
I think we are not using the above cases?

Reply:
no, otherwise it will crash loop. Just wanted to point out in the code.